lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -366,7 +366,7 @@ async sub transform_transaction {
         );
 
         my @transactions = await $self->_check_contract_transaction($transaction, $receipt);
-        push @transactions, $transaction;
+        push @transactions, $transaction if $amount->bgt(0);
```

Comment:
Would be nice if  this and line 468 both checked for "amountness" in the same way (I think they are both trying to do the same thing with a Math::BigFloat object?)

Reply:
@turnstep The amount mentioned in line #468 is using the `hash value` from $log->{data} and checking the validity of amount. And currently for amount in line 335 has already been transformed to `Math::BigFloat object`

Reply:
Line 468 is also a Math::BigFloat object:
```
$ pwd
/home/git/regentmarkets/perl-Net-Async-Blockchain

$ git diff -U1 | tail +5
@@ -467,2 +467,3 @@ async sub _check_contract_transaction {
             my $amount  = $self->get_valid_amount($log->{data});
+            printf STDERR "At line %d, amount is a %s\n", __LINE__, ref $amount;
             next unless $amount;

$ prove -vl t/eth.t 2>&1 | grep -m1 line
At line 468, amount is a Math::BigFloat
```

Reply:
@turnstep Yes `$self->get_valid_amount($log->{data});` will return as Math::BigFloat or undef. As we are converting the `$log->{data}` from hex value to Math::BigFloat or `undef`. We are returning undef when the amount is zero [here](https://github.com/binary-com/perl-Net-Async-Blockchain/blob/master/lib/Net/Async/Blockchain/ETH.pm#L653-L659). Since at line #468 is converting to `Math::BigFloat or undef(when amount is zero)` and at line #369 we have `Math::BigFloat` only to check the non-zero value.